__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,836 @@
+        });
+
+        test('returns message payload in truncate when message length over 40', () => {
+            runBasicMessageTests(PlainMessage.create('0'.repeat(45)), '0000000000000000000000000000000000000000...');
+        });
+
+        test('display mosaic icon when mosaics list contain non network mosaic', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: createMockTransferTransaction([new Mosaic(new MosaicId('461A9811A1DC1FBB'), UInt64.fromUint(0))]),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            jest.spyOn(vm, 'isIncomingTransaction').mockReturnValue(true);
+
+            // Act:
+            const amount = vm.getAmount();
+            const getAmountMosaicId = vm.getAmountMosaicId();
+
+            // Assert:
+            expect(amount).toBe(0);
+            expect(getAmountMosaicId).toBe(undefined);
+            expect(wrapper.find('.extend-icon-holder.mosaic-icon').exists()).toBe(true);
+        });
+
+        test('returns non network mosaic when transaction type is transfer transaction', () => {
+            // Arrange:
+            const mockMosaic = [
+                { idHex: networkMock.currency.mosaicIdHex, amount: 100 },
+                { idHex: '461A9811A1DC1FBB', amount: 100 },
+                { idHex: '119C8A107ACADD5F', amount: 100 },
+                { idHex: '2C56C7D764F17B09', amount: 100 },
+                { idHex: '5B2DCC92C3B0FFD8', amount: 100 },
+                { idHex: '4A1D5DA123186518', amount: 100 },
+            ];
+
+            const mosaics = mockMosaic.map((mosaic) => new Mosaic(new MosaicId(mosaic.idHex), UInt64.fromUint(mosaic.amount)));
+
+            runBasicNonMosaicListTests(createMockTransferTransaction(mosaics), [
+                { name: '461A9811A1DC1FBB', relativeAmount: '100' },
+                { name: '119C8A107ACADD5F', relativeAmount: '0.1' },
+                { name: '2C56C7D764F17B09', relativeAmount: '100' },
+                { name: '5B2DCC92C3B0FFD8', relativeAmount: '100' },
+                { name: '4A1D5DA123186518', relativeAmount: '100' },
+            ]);
+        });
+
+        test('returns empty non network mosaic when mosaics only contain network mosaic', () => {
+            // Arrange:
+            const mockMosaic = [{ idHex: networkMock.currency.mosaicIdHex, amount: 100 }];
+
+            const mosaics = mockMosaic.map((mosaic) => new Mosaic(new MosaicId(mosaic.idHex), UInt64.fromUint(mosaic.amount)));
+
+            runBasicNonMosaicListTests(createMockTransferTransaction(mosaics), []);
+        });
+
+        test('returns empty non network mosaic when transaction type is non transfer transaction', () => {
+            runBasicNonMosaicListTests(mockAccountAddressRestrictionTransaction(), []);
```

Comment:
rename mockAccountAddressRestrictionTransaction to createmMockAccountAddressRestrictionTransaction